src/osd/OSD.cc

```diff
@@ -1457,22 +1457,19 @@ void OSDService::send_incremental_map(epoch_t since, Connection *con,
            << " to " << con << " " << con->get_peer_addr() << dendl;
 
   MOSDMap *m = NULL;
-  while (!m) {
-    OSDSuperblock sblock(get_superblock());
-    if (since < sblock.get_oldest_map()) {
-      // just send latest full map
-      MOSDMap *m = new MOSDMap(monc->get_fsid(),
-			       osdmap->get_encoding_features());
-      m->cluster_osdmap_trim_lower_bound = sblock.cluster_osdmap_trim_lower_bound;
-      m->newest_map = sblock.get_newest_map();
-      get_map_bl(to, m->maps[to]);
-      send_map(m, con);
-      return;
-    }
-
+  OSDSuperblock sblock(get_superblock());
+  if (since < sblock.get_oldest_map()) {
```

Comment:
I think this patch PR is correct as is.

Going a bit further, I'm not actually sure why this if branch works this way.  build_incremental_map_msg handles the case (since < oldest_map) by sending oldest_map and then incrementals up to the current map.  That strikes me as probably the right behavior, so it might be that we can actually drop this if entirely.  I might be missing something though.

Reply:
> I think this patch is correct as is.
> 
> Going a bit further, I'm not actually sure why this if branch works this way. build_incremental_map_msg handles the case (since < oldest_map) by sending oldest_map and then incrementals up to the current map. That strikes me as probably the right behavior, so it might be that we can actually drop this if entirely. I might be missing something though.

Moreover, build_incremental_map_msg already adjusts `since` according to the trim_lower_bound epoch. Pushed cb675b13748c4d4588c38dee46eed114c0165201